src/async/async_api.c

```diff
@@ -144,7 +144,7 @@ __async_new_op_alloc(WT_SESSION_IMPL *session, const char *uri,
 	 */
 	if (op == NULL || op->state != WT_ASYNCOP_FREE) {
 		WT_STAT_CONN_INCR(session, async_full);
-		WT_RET(EBUSY);
+		return (EBUSY);
```

Comment:
I think a comment saying why we have a naked error return here would be helpful for the future so we know it was intentional.